### ISSUE TITLE
Unable to start jdbc connector

### ISSUE THREAD

**Issue by Susandhi** [confluentinc/kafka-connect-jdbc#209]
Hi... I'm trying to connect kafka to my database server but I'm getting this error please help me in solving this problem... here's the log

SLF4J: Found binding in [jar:file:/home/mustak/new_exp/confluent-3.2.0/share/java/kafka-serde-tools/slf4j-log4j12-1.7.6.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/mustak/new_exp/confluent-3.2.0/share/java/kafka-connect-elasticsearch/slf4j-simple-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/mustak/new_exp/confluent-3.2.0/share/java/kafka-connect-hdfs/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/mustak/new_exp/confluent-3.2.0/share/java/kafka-connect-s3/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/mustak/new_exp/confluent-3.2.0/share/java/kafka-connect-storage-common/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/mustak/new_exp/confluent-3.2.0/share/java/kafka/slf4j-log4j12-1.7.21.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
[2017-04-11 14:57:17,101] INFO StandaloneConfig values: 
	access.control.allow.methods = 
	access.control.allow.origin = 
	bootstrap.servers = [192.168.0.14:9092]
	internal.key.converter = class org.apache.kafka.connect.json.JsonConverter
	internal.value.converter = class org.apache.kafka.connect.json.JsonConverter
	key.converter = class io.confluent.connect.avro.AvroConverter
	offset.flush.interval.ms = 60000
	offset.flush.timeout.ms = 5000
	offset.storage.file.filename = /tmp/connect.offsets
	rest.advertised.host.name = null
	rest.advertised.port = null
	rest.host.name = null
	rest.port = 8083
	task.shutdown.graceful.timeout.ms = 5000
	value.converter = class io.confluent.connect.avro.AvroConverter
 (org.apache.kafka.connect.runtime.standalone.StandaloneConfig:180)
[2017-04-11 14:57:17,834] INFO Logging initialized @3107ms (org.eclipse.jetty.util.log:186)
[2017-04-11 14:57:19,433] INFO AvroConverterConfig values: 
	schema.registry.url = [http://192.168.0.14:8081]
	max.schemas.per.subject = 1000
 (io.confluent.connect.avro.AvroConverterConfig:169)
[2017-04-11 14:57:20,150] INFO AvroDataConfig values: 
	schemas.cache.config = 1000
	enhanced.avro.schema.support = false
	connect.meta.data = true
 (io.confluent.connect.avro.AvroDataConfig:169)
[2017-04-11 14:57:20,161] INFO AvroConverterConfig values: 
	schema.registry.url = [http://192.168.0.14:8081]
	max.schemas.per.subject = 1000
 (io.confluent.connect.avro.AvroConverterConfig:169)
[2017-04-11 14:57:20,162] INFO AvroDataConfig values: 
	schemas.cache.config = 1000
	enhanced.avro.schema.support = false
	connect.meta.data = true
 (io.confluent.connect.avro.AvroDataConfig:169)
[2017-04-11 14:57:20,261] INFO Kafka Connect starting (org.apache.kafka.connect.runtime.Connect:50)
[2017-04-11 14:57:20,262] INFO Herder starting (org.apache.kafka.connect.runtime.standalone.StandaloneHerder:72)
[2017-04-11 14:57:20,263] INFO Worker starting (org.apache.kafka.connect.runtime.Worker:119)
[2017-04-11 14:57:20,264] INFO Starting FileOffsetBackingStore with file /tmp/connect.offsets (org.apache.kafka.connect.storage.FileOffsetBackingStore:60)
[2017-04-11 14:57:20,273] INFO Worker started (org.apache.kafka.connect.runtime.Worker:124)
[2017-04-11 14:57:20,286] INFO Herder started (org.apache.kafka.connect.runtime.standalone.StandaloneHerder:74)
[2017-04-11 14:57:20,286] INFO Starting REST server (org.apache.kafka.connect.runtime.rest.RestServer:98)
[2017-04-11 14:57:21,555] INFO jetty-9.2.15.v20160210 (org.eclipse.jetty.server.Server:327)
Apr 11, 2017 2:57:28 PM org.glassfish.jersey.internal.Errors logErrors
WARNING: The following warnings have been detected: WARNING: The (sub)resource method listConnectors in org.apache.kafka.connect.runtime.rest.resources.ConnectorsResource contains empty path annotation.
WARNING: The (sub)resource method createConnector in org.apache.kafka.connect.runtime.rest.resources.ConnectorsResource contains empty path annotation.
WARNING: The (sub)resource method listConnectorPlugins in org.apache.kafka.connect.runtime.rest.resources.ConnectorPluginsResource contains empty path annotation.
WARNING: The (sub)resource method serverInfo in org.apache.kafka.connect.runtime.rest.resources.RootResource contains empty path annotation.

[2017-04-11 14:57:28,456] INFO Started o.e.j.s.ServletContextHandler@31198ceb{/,null,AVAILABLE} (org.eclipse.jetty.server.handler.ContextHandler:744)
[2017-04-11 14:57:28,589] INFO Started ServerConnector@306a944f{HTTP/1.1}{0.0.0.0:8083} (org.eclipse.jetty.server.ServerConnector:266)
[2017-04-11 14:57:28,608] INFO Started @13893ms (org.eclipse.jetty.server.Server:379)
[2017-04-11 14:57:28,618] INFO REST server listening at http://127.0.1.1:8083/, advertising URL http://127.0.1.1:8083/ (org.apache.kafka.connect.runtime.rest.RestServer:150)
[2017-04-11 14:57:28,624] INFO Kafka Connect started (org.apache.kafka.connect.runtime.Connect:56)
[2017-04-11 14:57:29,263] ERROR Failed to create job for source-mysql.properties (org.apache.kafka.connect.cli.ConnectStandalone:88)
[2017-04-11 14:57:29,270] ERROR Stopping after connector error (org.apache.kafka.connect.cli.ConnectStandalone:99)
java.util.concurrent.ExecutionException: org.apache.kafka.connect.runtime.rest.errors.BadRequestException: Connector configuration is invalid (use the endpoint `/{connectorType}/config/validate` to get a full list of errors)
	at org.apache.kafka.connect.util.ConvertingFutureCallback.result(ConvertingFutureCallback.java:80)
	at org.apache.kafka.connect.util.ConvertingFutureCallback.get(ConvertingFutureCallback.java:67)
	at org.apache.kafka.connect.cli.ConnectStandalone.main(ConnectStandalone.java:96)
Caused by: org.apache.kafka.connect.runtime.rest.errors.BadRequestException: Connector configuration is invalid (use the endpoint `/{connectorType}/config/validate` to get a full list of errors)
	at org.apache.kafka.connect.runtime.standalone.StandaloneHerder.putConnectorConfig(StandaloneHerder.java:161)
	at org.apache.kafka.connect.cli.ConnectStandalone.main(ConnectStandalone.java:93)
[2017-04-11 14:57:29,302] INFO Kafka Connect stopping (org.apache.kafka.connect.runtime.Connect:66)
[2017-04-11 14:57:29,303] INFO Stopping REST server (org.apache.kafka.connect.runtime.rest.RestServer:154)
[2017-04-11 14:57:29,360] INFO Stopped ServerConnector@306a944f{HTTP/1.1}{0.0.0.0:8083} (org.eclipse.jetty.server.ServerConnector:306)
[2017-04-11 14:57:29,439] INFO Stopped o.e.j.s.ServletContextHandler@31198ceb{/,null,UNAVAILABLE} (org.eclipse.jetty.server.handler.ContextHandler:865)
[2017-04-11 14:57:29,495] INFO REST server stopped (org.apache.kafka.connect.runtime.rest.RestServer:165)
[2017-04-11 14:57:29,495] INFO Herder stopping (org.apache.kafka.connect.runtime.standalone.StandaloneHerder:78)
[2017-04-11 14:57:29,495] INFO Worker stopping (org.apache.kafka.connect.runtime.Worker:131)
[2017-04-11 14:57:29,496] INFO Stopped FileOffsetBackingStore (org.apache.kafka.connect.storage.FileOffsetBackingStore:68)
[2017-04-11 14:57:29,497] INFO Worker stopped (org.apache.kafka.connect.runtime.Worker:151)
[2017-04-11 14:58:19,140] INFO Reflections took 58340 ms to scan 561 urls, producing 12993 keys and 85570 values  (org.reflections.Reflections:229)
[2017-04-11 14:58:19,336] INFO Herder stopped (org.apache.kafka.connect.runtime.standalone.StandaloneHerder:88)
[2017-04-11 14:58:19,337] INFO Kafka Connect stopped (org.apache.kafka.connect.runtime.Connect:71)

and this is my jdbc source properties
```
name=mysql-db-test
connector.class=io.confluent.connect.jdbc.JdbcSourceConnector
tasks.max=10
connection.url=jdbc:mysql://localhost/testdb?user=testuser&password=password
mode=incrementing
incrementing.column.name=id
topic.prefix=test-sqlite-jdbc-
```

**Comment by hriviere** [comment]
I got the same error with the mysql J connector (mysql-connector-java-5.0.8-bin.jar).
I replaced it by the maria db one (mariadb-java-client-1.5.9.jar) and it solved the issue

**Comment by ewencp** [comment]
This indicates that the configuration isn't valid, although @Susandhi it looks like you at least have all the required configs. We've actually fixed the issue where the issue with the config is not reported with the exception, but that is fixed in 0.10.2.1 which is still being voted on.

To determine what's going wrong you can also use the URL that is mentioned in the exception message to validate the config.

My guess would be that the driver issue @hriviere mentioned or something similar is the problem. Note that you need to add a driver for MySQL; the connector does not ship with that driver.

**Comment by Susandhi** [comment]
@hriviere @ewencp issue got resolved by using mariadb connector...
